File: C++ solutions/2405. Optimal Partition of String.cpp
class Solution {
public:
    int partitionString(string s) {
        map<char,int>m;
        int cnt=1;
        for(int i=0;i<s.size();i++){
            m[s[i]]++;
            if(m[s[i]]==2){
                cnt++;
                m.clear();
                m[s[i]]++;
            }
        }
        return cnt;
    }
};

File: JavaScript solutions/2405. Optimal Partition of String.js
/**
 * @param {string} s
 * @return {number}
 */
var partitionString = function (s) {
  let m = new Map();
  let cnt = 1;

  for (let i = 0; i < s.length; i++) {
    if (!m.has(s[i])) {
      m.set(s[i], 1);
    } else {
      cnt++;
      m.clear();
      m.set(s[i], 1);
    }
  }

  return cnt;
};
